packages/eslint-plugin/src/rules/no-duplicate-type-constituents.ts

```diff
@@ -111,6 +107,13 @@ export default createRule<Options, MessageIds>({
       const cachedTypeMap = new Map<Type, TSESTree.TypeNode>();
       node.types.reduce<TSESTree.TypeNode[]>(
         (uniqueConstituents, constituentNode) => {
+          const constituentNodeType = checker.getTypeAtLocation(
+            parserServices.esTreeNodeToTSNodeMap.get(constituentNode),
+          );
```

Comment:
```suggestion
          const constituentNodeType = parserServices.getTypeAtLocation(constituentNode);
```

`ParserServices` has a nice utility function to do this :)

Reply:
Nice! Thank you :) Fixed in [fc5d5d8](https://github.com/typescript-eslint/typescript-eslint/pull/9600/commits/fc5d5d827c435cb07259f2f3b707174d6d1daa71).